src/librustc_driver/driver.rs

```diff
@@ -667,7 +667,9 @@ pub fn phase_2_configure_and_expand<F>(sess: &Session,
                     new_path.push(path);
                 }
             }
-            env::set_var("PATH", &env::join_paths(new_path).unwrap());
+            if let Ok(p) = env::join_paths(new_path) {
```

Comment:
It would give an invalid PATH. Not sure this is what we want... At least, the current change doesn't set the variable at all if any path is not valid.